feature/qos/ate_tests/qos_basic_test/qos_basic_test.go

```diff
@@ -952,147 +944,194 @@ func ConfigureCiscoQos(t *testing.T, dut *ondatra.DUTDevice) {
-		name:        "dscp_based_classifier",
+		desc:        "classifier_ipv6_af3",
+		name:        "dscp_based_classifier_ipv6",
 		classType:   oc.Qos_Classifier_Type_IPV6,
-		termID:      "e_AF1_ipv6",
-		targetGroup: "e_AF1",
-		dscpSet:     []uint8{8, 9, 10, 11},
+		termID:      "4",
+		targetGroup: "target-group-AF3",
+		dscpSet:     []uint8{24, 25, 26, 27},
 	}, {
-		desc:        "classifier_ipv6_be0",
-		name:        "dscp_based_classifier",
+		desc:        "classifier_ipv6_af4",
+		name:        "dscp_based_classifier_ipv6",
 		classType:   oc.Qos_Classifier_Type_IPV6,
-		termID:      "f_BE0_ipv6",
-		targetGroup: "f_BE0",
-		dscpSet:     []uint8{4, 5, 6, 7},
+		termID:      "5",
+		targetGroup: "target-group-AF4",
+		dscpSet:     []uint8{32, 33, 34, 35},
 	}, {
-		desc:        "classifier_ipv6_be1",
-		name:        "dscp_based_classifier",
+		desc:        "classifier_ipv6_nc1",
+		name:        "dscp_based_classifier_ipv6",
 		classType:   oc.Qos_Classifier_Type_IPV6,
-		termID:      "g_BE1_ipv6",
-		targetGroup: "g_BE1",
-		dscpSet:     []uint8{0, 1, 2, 3},
+		termID:      "6",
+		targetGroup: "target-group-NC1",
+		dscpSet:     []uint8{48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59},
 	}}
 
-	t.Logf("qos Classifiers config: %v", classifiers)
-	queueName := []string{"a_NC1", "b_AF4", "c_AF3", "d_AF2", "e_AF1", "f_BE0", "g_BE1"}
-
-	for _, queue := range queueName {
-		q1 := q.GetOrCreateQueue(queue)
-		q1.Name = ygot.String(queue)
-
-	}
-	gnmi.Replace(t, dut, gnmi.OC().Qos().Config(), q)
-	for _, tc := range classifiers {
-		classifier := q.GetOrCreateClassifier(tc.name)
-		classifier.SetName(tc.name)
-		classifier.SetType(tc.classType)
-		term, err := classifier.NewTerm(tc.termID)
-		if err != nil {
-			t.Fatalf("Failed to create classifier.NewTerm(): %v", err)
-		}
+	t.Logf("qos Classifiers config cases: %v", cases)
+	for _, tc := range cases {
+		t.Run(tc.desc, func(t *testing.T) {
+			classifier := q.GetOrCreateClassifier(tc.name)
+			classifier.SetName(tc.name)
+			classifier.SetType(tc.classType)
+			term, err := classifier.NewTerm(tc.termID)
+			if err != nil {
+				t.Fatalf("Failed to create classifier.NewTerm(): %v", err)
+			}
 
-		term.SetId(tc.termID)
-		action := term.GetOrCreateActions()
-		action.SetTargetGroup(tc.targetGroup)
-		condition := term.GetOrCreateConditions()
-		if tc.classType == oc.Qos_Classifier_Type_IPV4 {
-			condition.GetOrCreateIpv4().SetDscpSet(tc.dscpSet)
-		} else if tc.classType == oc.Qos_Classifier_Type_IPV6 {
-			condition.GetOrCreateIpv6().SetDscpSet(tc.dscpSet)
-		}
-		fwdgroups := q.GetOrCreateForwardingGroup(tc.targetGroup)
-		fwdgroups.Name = ygot.String(tc.targetGroup)
-		fwdgroups.OutputQueue = ygot.String(tc.targetGroup)
-		gnmi.Replace(t, dut, gnmi.OC().Qos().Config(), q)
+			term.SetId(tc.termID)
+			action := term.GetOrCreateActions()
+			action.SetTargetGroup(tc.targetGroup)
+			condition := term.GetOrCreateConditions()
+			if tc.name == "dscp_based_classifier_ipv4" {
```

Comment:
check for type instead of name? same previous PRs

Reply:
The original tests function which ConfigureQoS had similar way to differentiate between v4 and v6 conditions was with comparing with string.
condition := term.GetOrCreateConditions()
if tc.name == "dscp_based_classifier_ipv4" {
condition.GetOrCreateIpv4().SetDscpSet(tc.dscpSet)
} else if tc.name == "dscp_based_classifier_ipv6" {
condition.GetOrCreateIpv6().SetDscpSet(tc.dscpSet)
}

But since in the previous releases , we did not support different policy-map name for V4 or v6 under same , I used type to modify the condition. Since in this release , we support the same, changed the test back how the original test was written.
Please let me know if we still have to change the test by using the type to derive the condition , I can do the same

Reply:
fixed it